src/player.ts

```diff
@@ -96,209 +74,54 @@ export class Player {
-        }
-        if (wallCollision.right) {
-          this.x = this.canvas.width - this.width / 2;
-        } else {
-          this.x += this.diagonalSpeed;
-        }
-      }
-    } else if (keysPressed['ArrowDown'] && keysPressed['ArrowLeft']) {
-      this.facing = FacingAngles.BottomLeft;
-      if (enemyCollision) {
-        this.y -= this.diagonalSpeed * 2;
-        this.x += this.diagonalSpeed * 2;
-        this.takeDamage(10, true);
-      } else {
-        if (wallCollision.bottom) {
-          this.y = this.canvas.height - this.height / 2;
-        } else {
-          this.y += this.diagonalSpeed;
-        }
-        if (wallCollision.left) {
-          this.x = this.width / 2;
-        } else {
-          this.x -= this.diagonalSpeed;
-        }
-      }
-    } else if (keysPressed['ArrowDown'] && keysPressed['ArrowRight']) {
-      this.facing = FacingAngles.BottomRight;
-      if (enemyCollision) {
-        this.y -= this.diagonalSpeed * 2;
-        this.x -= this.diagonalSpeed * 2;
-        this.takeDamage(10, true);
-      } else {
-        if (wallCollision.bottom) {
-          this.y = this.canvas.height - this.height / 2;
-        } else {
-          this.y += this.diagonalSpeed;
-        }
-        if (wallCollision.right) {
-          this.x = this.canvas.width - this.width / 2;
-        } else {
-          this.x += this.diagonalSpeed;
-        }
-      }
-    } else if (keysPressed['ArrowUp']) {
-      this.facing = FacingAngles.Top;
-      if (enemyCollision) {
-        this.y += this.speed * 2;
-        this.takeDamage(10, true);
-      } else if (wallCollision.top) {
-        this.y = this.height / 2;
-      } else {
-        this.y -= this.speed;
-      }
-    } else if (keysPressed['ArrowDown']) {
-      this.facing = FacingAngles.Bottom;
-      if (enemyCollision) {
-        this.y -= this.speed * 2;
-        this.takeDamage(10, true);
-      } else if (wallCollision.bottom) {
-        this.y = this.canvas.height - this.height / 2;
-      } else {
-        this.y += this.speed;
-      }
-    } else if (keysPressed['ArrowLeft']) {
-      this.facing = FacingAngles.Left;
-      if (enemyCollision) {
-        this.x += this.speed * 2;
-        this.takeDamage(10, true);
-      } else if (wallCollision.left) {
-        this.x = this.width / 2;
-      } else {
-        this.x -= this.speed;
-      }
-    } else if (keysPressed['ArrowRight']) {
-      this.facing = FacingAngles.Right;
-      if (enemyCollision) {
-        this.x -= this.speed * 2;
-        this.takeDamage(10, true);
-      } else if (wallCollision.right) {
-        this.x = this.canvas.width - this.width / 2;
-      } else {
-        this.x += this.speed;
-      }
-    }
-  }
-
-  handleAttack(keysPressed: { [key: string]: boolean }, enemies: Array<Enemy>) {
-    if (
-      keysPressed['mousedown'] &&
-      Date.now() - this.attackTimer >= this.weapon.cooldown
-    ) {
-      this.attack();
+      attackHandler(this, enemies);
```

Comment:
Why does this need to be called here and in playerController? I think ideally it would only be called in the controller class.

Reply:
The call in the controller only initiates the attack while the player.update is called every game tick to account for changes in weapon position.

Reply:
Fixed. Separated attack initiation and attack update into own functions. 82a6944